### ISSUE TITLE
creating an identity for external ottca missing CA ID property

### ISSUE THREAD

**Issue by qrkourier** [openziti/ziti-console#151]
Console v2.8.2 doesn't set `POST` create request property `enrollment.ottca: 5L5JFo5dxKUFbiNXohMPll` to the ID of the selected external CA.

![image](https://github.com/openziti/ziti-console/assets/1434400/d9ea3e75-2fdf-4897-ab6b-d1f1b17e1f3d)

This results in an identity that can not then be enrolled with a client certificate from the selected external CA. The defunct created identity has property `enrollment: {}` instead of the following excerpt.

```json
            "enrollment": {
                "ottca": {
                    "ca": {
                        "_links": {
                            "jwt": {
                                "href": "./cas/5L5JFo5dxKUFbiNXohMPll/jwt"
                            },
                            "self": {
                                "href": "./cas/5L5JFo5dxKUFbiNXohMPll"
                            }
                        },
                        "entity": "cas",
                        "id": "5L5JFo5dxKUFbiNXohMPll",
                        "name": "magenta"
                    },
                    "caId": "5L5JFo5dxKUFbiNXohMPll",
                    "expiresAt": "2023-07-18T17:09:03.031Z",
                    "id": "I5NtczBOun",
                    "jwt": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbSI6Im90dGNhIiwiZXhwIjoxNjg5NzAwMTQzLCJpc3MiOiJodHRwczovL3ppdGktZWRnZS1jb250cm9sbGVyOjEyODAiLCJqdGkiOiJjM2FmODA0Yy0yOGMwLTQxNzgtYTYwMy1lNGJhYmQ5MmJmZTUiLCJzdWIiOiJJNU50Y3pCb3VsIn0.BK2-oTTl0VxxTcTJJM0oqPlt46_x-OqSjWeXUhJ-paSHXJN8dwUVrMLk6GNK7Kx7tgigxhRvS9wgHiH9QbETx7ngaQ4MiPZSNGNL6SHUq9RD3CJqlJ5IIZ1jvsWIeYXBAkqfnoo0b4dyQjB9i6jgne9bCeO0C0TbotNWGDJgaIENsIv-ElnH2puqBHkIA8bCBE_4d8KQ3KbnIhD_2hw9LoKjnh_qg5OjT8MznfbYPRIxvrYG3wryDwaP3XGr4PoYFbfx2WByz1MtT6OwMgAnO0ZHYZKhW6-udya2_Ripy7wTo574kC3als58ePE5x_mKf11tRWUTvpvkeEqfyvsaJugviYuaFo19nufNVeEYG0RR1Sy3N1UeakCj1wtR-jSBxHdwgcb65apFOsfAUA1G7wgkA0UrmHQ5OwOVfWwipcaveHLizArnH-TPjEh_cbt5dprWm_3OuArPbSiySDNYBbmIUOPeczly5LR__h_gThceV5Y4nuWApEPRMZ04SteD_EQG3Do5NiB7KXQhgVDsE2SD8c64uxUjK9kpOrVdZMgaW5giPy2aBaAX_VWQdbj9nDWBBqvvZbrih7m_rNCdHHohvrjLNbfcNEFYWY_ocJHtABVzyZceBcupiUPIT8AgKmTG4NCP9c_9GhCVdXO1z6zB90yz6lu4h8C87PpZxq0",
                    "token": "c3af804c-28c0-4178-a603-e4babd92bfe5"
                }
            },
```

A correctly-created identity for `ottca` enrollment will generate an outstanding enrollment entity like this.

```json
        {
            "_links": {
                "self": {
                    "href": "./enrollments/I5NtczBOun"
                }
            },
            "createdAt": "0001-01-01T00:00:00.000Z",
            "id": "I5NtczBOun",
            "tags": {},
            "updatedAt": "0001-01-01T00:00:00.000Z",
            "caId": "5L5JFo5dxKUFbiNXohMPll",
            "expiresAt": "2023-07-18T17:09:03.031Z",
            "identity": {
                "_links": {
                    "auth-policies": {
                        "href": "./auth-policies/default"
                    },
                    "authenticators": {
                        "href": "./identities/I5NtczBoul/authenticators"
                    },
                    "edge-router-policies": {
                        "href": "./identities/I5NtczBoul/edge-router-policies"
                    },
                    "edge-routers": {
                        "href": "./identities/I5NtczBoul/edge-routers"
                    },
                    "enrollments": {
                        "href": "./identities/I5NtczBoul/enrollments"
                    },
                    "failed-service-requests": {
                        "href": "./identities/I5NtczBoul/failed-service-requests"
                    },
                    "posture-data": {
                        "href": "./identities/I5NtczBoul/posture-data"
                    },
                    "self": {
                        "href": "./identities/I5NtczBoul"
                    },
                    "service-configs": {
                        "href": "./identities/I5NtczBoul/service-configs"
                    },
                    "service-policies": {
                        "href": "./identities/I5NtczBoul/service-policies"
                    },
                    "services": {
                        "href": "./identities/I5NtczBoul/services"
                    }
                },
                "entity": "identities",
                "id": "I5NtczBoul",
                "name": "kentest24"
            },
            "identityId": "I5NtczBoul",
            "jwt": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbSI6Im90dGNhIiwiZXhwIjoxNjg5NzAwMTQzLCJpc3MiOiJodHRwczovL3ppdGktZWRnZS1jb250cm9sbGVyOjEyODAiLCJqdGkiOiJjM2FmODA0Yy0yOGMwLTQxNzgtYTYwMy1lNGJhYmQ5MmJmZTUiLCJzdWIiOiJJNU50Y3pCb3VsIn0.BK2-oTTl0VxxTcTJJM0oqPlt46_x-OqSjWeXUhJ-paSHXJN8dwUVrMLk6GNK7Kx7tgigxhRvS9wgHiH9QbETx7ngaQ4MiPZSNGNL6SHUq9RD3CJqlJ5IIZ1jvsWIeYXBAkqfnoo0b4dyQjB9i6jgne9bCeO0C0TbotNWGDJgaIENsIv-ElnH2puqBHkIA8bCBE_4d8KQ3KbnIhD_2hw9LoKjnh_qg5OjT8MznfbYPRIxvrYG3wryDwaP3XGr4PoYFbfx2WByz1MtT6OwMgAnO0ZHYZKhW6-udya2_Ripy7wTo574kC3als58ePE5x_mKf11tRWUTvpvkeEqfyvsaJugviYuaFo19nufNVeEYG0RR1Sy3N1UeakCj1wtR-jSBxHdwgcb65apFOsfAUA1G7wgkA0UrmHQ5OwOVfWwipcaveHLizArnH-TPjEh_cbt5dprWm_3OuArPbSiySDNYBbmIUOPeczly5LR__h_gThceV5Y4nuWApEPRMZ04SteD_EQG3Do5NiB7KXQhgVDsE2SD8c64uxUjK9kpOrVdZMgaW5giPy2aBaAX_VWQdbj9nDWBBqvvZbrih7m_rNCdHHohvrjLNbfcNEFYWY_ocJHtABVzyZceBcupiUPIT8AgKmTG4NCP9c_9GhCVdXO1z6zB90yz6lu4h8C87PpZxq0",
            "method": "ottca",
            "token": "c3af804c-28c0-4178-a603-e4babd92bfe5"
        }
```

**Comment by qrkourier** [comment]
Full create POST request example:

```json
{
  "name": "kentest24",
  "type": "Device",
  "isAdmin": false,
  "appData": {},
  "authPolicyId": "default",
  "defaultHostingCost": 0,
  "defaultHostingPrecedence": "required",
  "enrollment": {
    "ott": false,
    "ottca": "5L5JFo5dxKUFbiNXohMPll",
    "updb": ""
  },
  "externalId": "",
  "roleAttributes": [],
  "serviceHostingCosts": {},
  "serviceHostingPrecedences": {},
  "tags": {}
}
```

**Comment by JeremyTellier** [comment]
Fixed in 2.8.4
